File: 0729-my-calendar-i/0729-my-calendar-i.py
class MyCalendar(object):
    
    def __init__(self):
        self.cal=[]
        

    def book(self, start, end):
        for [s,e] in self.cal:
            if start<e and end>s:
                return False
        self.cal.append([start,end])
        return True
        


# Your MyCalendar object will be instantiated and called as such:
# obj = MyCalendar()
# param_1 = obj.book(start,end)

File: 0729-my-calendar-i/0729-my-calendar-i.cpp
class MyCalendar {
    vector<vector<int>> cal;
public:
    MyCalendar() {  }

    bool book(int start, int end) {
        for (auto it : cal) {
            if (it[0] <= start && start < it[1] ||
                it[0] < end && end <= it[1] || start<=it[0]&&it[1]<=end) {
                return false;
            }
        }
        cal.push_back({start, end});
        return true;
    }
};

/**
 * Your MyCalendar object will be instantiated and called as such:
 * MyCalendar* obj = new MyCalendar();
 * bool param_1 = obj->book(start,end);
 */